src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -355,19 +668,573 @@ private CategoryDataset buildDataSet(StaplerRequest req) {
         return dsb.build();
     }
 
-    private JFreeChart createChart(StaplerRequest req,CategoryDataset dataset) {
+    /**
+     * Utility method to count the number of failed/passed/skipped testcases.
+     * @param tests List of failed/passed/skipped testcases.
+     * @param allPackages True if all packages needs to considered else false.
+     * @param projectLevel The package level that needs to be considered.
+     * @param dataset Data structure for storing the number of failed/passed/skipped testcases which will be used to
+     *            generate trends.
+     * @param testResultAction Current build.
+     * @param seriesName Series name as "failed"/"skipped"/"total" for plotting.
+     */
+    private void buildDataSetPerProjectUtil (List<CaseResult> tests, boolean allPackages, String projectLevel,
+                                             DataSetBuilder<String, NumberOnlyBuildLabel> dataset,
+                                             AbstractTestResultAction<?> testResultAction, String seriesName) {
+        String toolTipString = "";
+        int maxToolTipLength = 100;
+        boolean generateToolTip = true;
+        int count = 0;
+        for (CaseResult caseResult : tests) {
+            String caseName = caseResult.getFullName();
+            if (!allPackages && !caseName.startsWith(projectLevel))
+                continue;
+            count++;
```

Comment:
give a meaningful name to this count var

Reply:
Done.